tests/test_MomentsClassifier.py

```diff
@@ -56,6 +56,20 @@ def testSingleFramePlugin(self):
         for ii in range(self.n_stars, self.n_stars + self.n_gals):
             self.assertGreater(catalog[ii].get("base_ClassificationSizeExtendedness_value"), 0.02)
 
+    def testFailure(self):
+        """Test that MeasurementError is raised properly."""
+        config = measBase.SingleFrameMeasurementConfig()
+        task = self.makeSingleFrameMeasurementTask(config=config)
+        exposure, catalog = self.dataset.realize(10.0, task.schema, randomSeed=5)
+        plugin = task.plugins["base_ClassificationSizeExtendedness"]
+        plugin_order = plugin.getExecutionOrder()
+        task.run(catalog, exposure, endOrder=plugin_order)
+        catalog["slot_Shape_flag"] = 1
+        for record in catalog:
+            with self.subTest(id=record["id"]):
+                with self.assertRaises(measBase.MeasurementError):
```

Comment:
Please use `assertRaisesRegex` to confirm that the error you expected is what is being raised. I think just "Shape flag is set." is enough.